common/types/signatures.go

```diff
@@ -27,11 +28,63 @@ func (s *EdSignature) DecodeScale(decoder *scale.Decoder) (int, error) {
 
 // String returns a string representation of the Signature, for logging purposes.
 // It implements the Stringer interface.
-func (s *EdSignature) String() string {
+func (s EdSignature) String() string {
 	return hex.EncodeToString(s.Bytes())
 }
 
 // Bytes returns the byte representation of the Signature.
 func (s *EdSignature) Bytes() []byte {
+	if s == nil {
+		return nil
+	}
 	return s[:]
 }
+
+type VrfSignature [VrfSignatureSize]byte
+
+// EmptyVrfSignature is a canonical empty VrfSignature.
+var EmptyVrfSignature VrfSignature
+
+// String returns a string representation of the Signature, for logging purposes.
+// It implements the Stringer interface.
+func (s VrfSignature) String() string {
+	return hex.EncodeToString(s.Bytes())
+}
+
+// Bytes returns the byte representation of the Signature.
+func (s *VrfSignature) Bytes() []byte {
+	if s == nil {
+		return nil
+	}
+	return s[:]
+}
+
+// Cmp compares s and x and returns:
+//
+//	-1 if s <  x
+//	 0 if s == x
+//	+1 if s >  x
+//
+// The comparison is done in little endian order.
+// Additionally, if x is nil, -1 is returned.
+func (s *VrfSignature) Cmp(x *VrfSignature) int {
+	if x == nil {
+		return -1
+	}
+
+	// VRF signatures are little endian, so we need to compare in reverse order.
+	for i := len(s) - 1; i >= 0; i-- {
+		if s[i] < x[i] {
+			return -1
+		}
+		if s[i] > x[i] {
+			return 1
+		}
+	}
+	return 0
+}
+
+// LSB returns the least significant byte of the signature.
+func (s *VrfSignature) LSB() byte {
```

Comment:
since we are relying on this for coinflip. is it worth writing a sanity check test that verifies that out of 10K signatures, we at least get a 30/70 or 40/60 of different result?

Reply:
sure, should be simple enough 🙂